src/vm/exceptionhandling.cpp

```diff
@@ -4585,6 +4594,29 @@ VOID DECLSPEC_NORETURN DispatchManagedException(PAL_SEHException& ex)
     }
 }
 
+VOID PALAPI HandleHardwareException(PAL_SEHException* ex)
+{
+    // Create frame necessary for the exception handling
+    FrameWithCookie<FaultingExceptionFrame> fef;
+#if defined(WIN64EXCEPTIONS)
+    *((&fef)->GetGSCookiePtr()) = GetProcessGSCookie();
+#endif // WIN64EXCEPTIONS
+    GCX_COOP();
+    fef.InitAndLink(&ex->ContextRecord);
+
+    // A hardware exception is handled only if it happened in a jitted code or 
+    // in one of the JIT helper functions (JIT_MemSet, ...)
+    PCODE controlPc = GetIP(&ex->ContextRecord);
+    if (ExecutionManager::IsManagedCode(controlPc) || IsIPInJitHelper(controlPc))
```

Comment:
It moves the context to the first managed frame, and then moves the IP backwards by STACKWALK_CONTROLPC_ADJUST_OFFSET. I do not think we have the move backward part there today.